src/com/goide/inspections/GoPlaceholderCountInspection.java

```diff
@@ -16,170 +16,261 @@
 
-    String placeholderText = resolve(argument);
-    if (placeholderText == null) return -1;
+        String functionName = StringUtil.toLowerCase(((GoFunctionOrMethodDeclaration)resolved).getName());
+        if (functionName == null) return;
 
-    Matcher placeholders = PLACEHOLDER_PATTERN.matcher(placeholderText);
-    Matcher countPlaceholders;
+        if (GoPlaceholderChecker.isFormattingFunction(functionName)) {
+          checkPrintf(holder, o, (GoFunctionOrMethodDeclaration)resolved);
+        }
+        else if (GoPlaceholderChecker.isPrintingFunction(functionName)) {
+          checkPrint(holder, o, (GoFunctionOrMethodDeclaration)resolved);
+        }
+      }
+    };
+  }
 
-    int indexCount = 0;
-    int maxIndexCount = 0;
+  private static void checkPrint(
+    @NotNull ProblemsHolder holder,
+    @NotNull GoCallExpr callExpr,
+    @NotNull GoFunctionOrMethodDeclaration declaration) {
 
-    while (placeholders.find()) {
-      indexCount++;
+    List<GoExpression> arguments = callExpr.getArgumentList().getExpressionList();
+    if (arguments.isEmpty()) return;
 
-      countPlaceholders = INDEXED_PLACEHOLDER_PATTERN.matcher(placeholders.group());
-      while (countPlaceholders.find()) {
-        int index = StringUtil.parseInt(countPlaceholders.group().replaceAll("\\[|\\]", ""), -1);
-        if (index > 0 && index != indexCount) {
-          indexCount = index;
+    GoExpression firstArg = arguments.get(0);
+    if (GoTypeUtil.isString(firstArg.getGoType(null))) {
+      String firstArgText = GoPlaceholderChecker.resolve(firstArg);
+      if (firstArgText != null) {
+        if (GoPlaceholderChecker.hasPlaceholder(firstArgText)) {
+          String message = "Possible formatting directive in <code>#ref</code>";
+          holder.registerProblem(firstArg, message, ProblemHighlightType.GENERIC_ERROR_OR_WARNING);
+          return;
         }
-        if (maxIndexCount < indexCount) {
-          maxIndexCount = indexCount;
+      }
+    }
+
+    // TODO florin: Check first argument for os.Std* output
+    // Ref code: https://github.com/golang/go/blob/79f7ccf2c3931745aeb97c5c985b6ac7b44befb4/src/cmd/vet/print.go#L617
+
+    String declarationName = declaration.getName();
+    boolean isLn = declarationName != null && declarationName.endsWith("ln");
+    for (GoExpression argument : arguments) {
+      GoType goType = argument.getGoType(null);
+      if (isLn && GoTypeUtil.isString(goType)) {
+        String argText = GoPlaceholderChecker.resolve(argument);
+        if (argText != null && argText.endsWith("\\n")) {
+          String message = "Function already ends with new line";
+          holder.registerProblem(argument, message, ProblemHighlightType.GENERIC_ERROR_OR_WARNING);
         }
       }
-      if (maxIndexCount < indexCount) {
-        maxIndexCount = indexCount;
+      else if (GoTypeUtil.isFunction(goType)) {
+        String message = "Argument <code>#ref</code> is not a function call";
```

Comment:
It doesn't.

<img width="957" alt="screenshot 2016-06-15 15 00 53" src="https://cloud.githubusercontent.com/assets/140920/16078898/030e9732-330a-11e6-924b-a763571282f2.png">